### PR TITLE
RHOL-1120: Fix PrettyPrinter output for maps

### PR DESCRIPTION
**Pull request by tymm**
## Overview
Output was messed up because we `zipWithIndex`ed over a `HashMap`.

### JIRA ticket:
https://rchain.atlassian.net/browse/RHOL-1120

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala [hunk 1]
```diff
@@ -91,7 +91,7 @@ case class PrettyPrinter(
       case ESetBody(ParSet(pars, _, _, remainder)) =>
         pure("Set(") |+| buildSeq(pars.sortedPars) |+| buildRemainderString(remainder) |+| pure(")")
       case EMapBody(ParMap(ps, _, _, remainder)) =>
-        pure("{") |+| (pure("") /: ps.sortedMap.zipWithIndex) {
+        pure("{") |+| (pure("") /: ps.sortedList.zipWithIndex) {
```
Comment (ArturGajowy) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala]: Please make the sortedMap private to prevent similar errors :)

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala [hunk 2]
```diff
@@ -101,6 +101,35 @@ class CollectPrinterSpec extends FlatSpec with Matchers {
     result shouldBe """{x0 : x1, 7 : "Seven"...free0}"""
   }
 
+  "Map" should "Print commas correctly" in {
```
Comment (ArturGajowy) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala]: What does this have to do with commas? I thought it's about the element (key) order?

**Comment by tymm** [comment]
It is refering to a previous PR: https://github.com/rchain/rchain/pull/1823

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala (hunk 2 above)
Comment (tymm) [reply rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala]: True, but it had the effect that commas could be messed up and this is how we actually discovered the bug.

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala (hunk 1 above)
Comment (tymm) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala]: Good idea. Done.

Comment (tymm) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala]: Or actually not done. More changes required elsewhere.

File: models/src/test/scala/coop/rchain/models/SortedParMapSpec.scala [hunk 3]
```diff
@@ -32,7 +32,7 @@ class SortedParMapSpec extends FlatSpec with PropertyChecks with Matchers {
   )
 
   private def roundTripTest(parMap: SortedParMap): Assertion =
-    EMap.parseFrom(serializeEMap(parMap)) should ===(EMap(parMap.sortedMap.map(toKVpair).toSeq))
+    EMap.parseFrom(serializeEMap(parMap)) should ===(EMap(parMap.sortedList.map(toKVpair)))
```
Comment (ArturGajowy) [inline models/src/test/scala/coop/rchain/models/SortedParMapSpec.scala]: Sorry for the avalanche of changes, but...

serializeEMap is defined in this test so it isn't really testing a concept from production code. Instead, I believe the definition of that method should be using the ParMapTypMapper to convert the ParMap to EMap and do `toByteArray` from there. Good to merge after that change :)

Comment (tymm) [reply models/src/test/scala/coop/rchain/models/SortedParMapSpec.scala]: The methods in `ParMapTypeMapper` only work on `ParMap` while we want to test `SortedParMap` here.

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala (hunk 1 above)
Comment (tymm) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala]: Now it should be done.
